third-party/cmake/DownloadAndExtractFile.cmake

```
@@ -74,11 +74,26 @@ list(GET status 0 status_code)
 list(GET status 1 status_string)
 
 if(NOT status_code EQUAL 0)
-  message(FATAL_ERROR "error: downloading '${URL}' failed
+  # Retry on certain errors, e.g. CURLE_COULDNT_RESOLVE_HOST, which is often
+  # seen with libtermkey (www.leonerd.org.uk).
+  if(status_code EQUAL 6)  # "Couldn't resolve host name"
+    message(STATUS "error: downloading '${URL}' failed (${status_string}, status ${status_code}) - retrying in 10s...")
```

avoid "error" or "fail" in non-fatal messages. On Windows it could even cause a failed build, because MSVC looks at the messages, it can't rely on error codes. Also avoids noise when grepping the log.

```suggestion
    message(STATUS "warning: retrying '${URL}' (${status_string}, status ${status_code})")
```

---

:+1: 
Let's keep the " - retrying in 10s.." though?

---

why?

---

Oh, it's at the start.. :+1: